jre_emul/android/platform/libcore/ojluni/src/main/java/java/util/zip/ZipUtils.java

```diff
@@ -0,0 +1,211 @@
+/*
+ * Copyright (c) 2013, 2015, Oracle and/or its affiliates. All rights reserved.
+ * DO NOT ALTER OR REMOVE COPYRIGHT NOTICES OR THIS FILE HEADER.
+ *
+ * This code is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License version 2 only, as
+ * published by the Free Software Foundation.  Oracle designates this
+ * particular file as subject to the "Classpath" exception as provided
+ * by Oracle in the LICENSE file that accompanied this code.
+ *
+ * This code is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+ * version 2 for more details (a copy is included in the LICENSE file that
+ * accompanied this code).
+ *
+ * You should have received a copy of the GNU General Public License version
+ * 2 along with this work; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ */
+
+package java.util.zip;
+
+//import java.nio.file.attribute.FileTime;
+import java.util.Date;
+import java.util.concurrent.TimeUnit;
+
+public
+class ZipUtils {
+
+    /**
+     * J2ObjC added: avoids using java.nio.file.attribute.FileTime,
+     * instead implements FileTime class with relevant methods.
+     */
+    public static class FileTime implements Comparable<FileTime>{
```

Comment:
As we discussed, please move this inner class to a new source file in jre_emul/Classes/com/google/j2objc/nio/file/attribute/FileTime.java, which will use the Apache license and may include you as its @author in the class doc-comment. This reduces the delta on ZipUtils.java to just the changed import statement above.

Reply:
I think I already moved this class and changed corresponding licenses in my most recent commit.